ci/scripts/ccache_setup.sh

```diff
@@ -23,4 +23,4 @@ echo "ARROW_USE_CCACHE=ON" >> $GITHUB_ENV
 echo "CCACHE_COMPILERCHECK=content" >> $GITHUB_ENV
 echo "CCACHE_COMPRESS=1" >> $GITHUB_ENV
 echo "CCACHE_COMPRESSLEVEL=6" >> $GITHUB_ENV
-echo "CCACHE_MAXSIZE=500M" >> $GITHUB_ENV
+echo "CCACHE_MAXSIZE=2G" >> $GITHUB_ENV
```

Comment:
Won't this produce a lot of evictions if the total allowed size of caches for a given project is 10GB?

Reply:
That is, if each cache nears 2 GB, then we can't store more than 5 caches at once, which is too small given the number of ccache-enabled CI configurations.

Reply:
cc @kszucs for potential opinions

Reply:
We seem to have a special allowance (or its part of Enterprise) as we have 80G of active caches. Regardless of that, we are forced to create a unique key on each run to update the cache due to the shortcomings of `actions/cache` so we will have a bunch of unused caches that will get evicted anyway, which is not ideal but we currently have no better solution. 

This is why I now think that we should replace ccache with sccache not only in crossbow but also in arrow. (given that it  works well, I am already working on getting a test setup in crossbow)

Reply:
> Yes, but what I mean is that if caches get evicted too often, then we won't benefit from caching :-)

Yes, I understand but with the workaround to force `actions/cache` to update (and even before due to the key being pretty restrictive), we will have a lot of evictions. 

We could use an even bigger size and try to share ccaches between workflows by using similar keys, that way even if we have a lot of evictions the chance of a cache hit would still be high. Or would this cause issues with ccache?

Reply:
> Some additional info: In my fork with cold cache the new cache would be around 750M. So maybe 1G as a compromise for now

Wow. Ok, then 1GB sounds like a fine compromise.

Reply:
> We could use an even bigger size and try to share ccaches between workflows by using similar keys, that way even if we have a lot of evictions the chance of a cache hit would still be high.

As soon as the compiler or compilation options change then cache entries are not reused. So I don't think that would work.

Reply:
(generally speaking, ccache hashes a lot of things in its cache keys so to avoid false positives, i.e. situations where you would not get the expected compilation result)

Reply:
Yes, sccache would be a better solution IMHO. But it will longer to deploy than these simple fixes.

Reply:
My idea of a "global" cache would not work, as it turns out the caches also have an os dependent version (maybe dependent on something else too? who knows, it's undocumented 💢 ): https://github.com/actions/cache/issues/362 

I'll set the size to 1G for now.